crates/ruff_python_ast/src/visibility.rs

```diff
@@ -131,11 +131,9 @@ fn stem(path: &str) -> &str {
 
 /// Return the `Visibility` of the Python file at `Path` based on its name.
 pub fn module_visibility(path: &Path) -> Visibility {
```

Comment:
I think we may be able to look at `module_path` here rather than `Path`:

```
/// Create a module path from a (package, path) pair.
///
/// For example, if the package is `foo/bar` and the path is `foo/bar/baz.py`,
/// the call path is `["baz"]`.
pub fn to_module_path(package: &Path, path: &Path) -> Option<Vec<String>> {
  ...
}
```

(The only caller of this already has `module_path` in scope.)

I think that would solve the same problem, but would use our centralized module detection logic rather than implementing additional `__init__.py` checks here. Can you take a look and see if that works?